HAR-6861 - Käsittlee Vesiväylien suunnitellut työt vs hinnoittelu-ui huomioiden alku/loppuajat

erno opened the pull request.

Commit:
valmistele suunniteltujen töiden pvm-käsittelyn porttaus vesiväylät -> kanavat

Täälläkin pitäisi huomioida urakkavuodet ja mätsätä suunnitellun
työn voimassaoloajat sen mukaan

Commit:
Merge remote-tracking branch 'origin/develop' into HAR-6810

Commit:
Ota huomioon suunniteltujen töiden päivämäärät

Commit:
poista tarpeeton and

Commit:
Paranna hinnoittelun vs suunniteltujen töiden toimintaa

- Korjaa yksikköhinnan laskenta (otetaan suunnittelusta eikä tpk:n tiedoista)
- Tarjoa valittavaksi vain nykyisen urakkakauden suunniteltuja toimenpiteitä
- Laske hinta oikein nykyisen urakkakauden suunnittelun hinnalla

Kauko committed:
Merge branch 'develop' into HAR-6810

Kauko commented on src/cljs/harja/tiedot/vesivaylat/urakka/toimenpiteet/yksikkohintaiset.cljs:
```diff
@@ -527,8 +527,10 @@
           toimenpiteen-oma-hinnoittelu (::to/oma-hinnoittelu hinnoiteltava-toimenpide)
           hinnat (or (::h/hinnat toimenpiteen-oma-hinnoittelu) [])
           tyot (or (::h/tyot toimenpiteen-oma-hinnoittelu) [])]
+      (log "ATH: pvm" (pr-str (::to/pvm hinnoiteltava-toimenpide)))
```
Ei oo varmaan tarkoituksella täällä

Kauko commented on src/cljs/harja/views/vesivaylat/urakka/toimenpiteet/toimenpiteen_hinnoittelu.cljs:
```diff
@@ -168,40 +169,57 @@
+(defn- suunniteltu-tyo-voimassa-paivamaaralle? [tp-pvm tyo]
+  (assert some? tp-pvm)
+  (log "onko pvm" (pr-str  (pvm/pvm? tp-pvm)) )
+  (log "TPH: valissa: " (pr-str  tp-pvm) (:alkupvm tyo) (:loppupvm tyo) "->" (pr-str (pvm/valissa? tp-pvm (:alkupvm tyo) (:loppupvm tyo))))
```
Tästäkin lokitukset pois?

Commit:
siivoa lokitukset

Commit:
korjaa testi

Kauko committed:
Merge branch 'develop' into HAR-6810

Kauko committed:
Piilota hinnoittelussa työn vuosiluku

Vuosilukua ehdotettiin lisättäväksi pudotusvalikkoon, kun ajateltiin,
että valikkoon voi tulla suunniteltuja hintoja eri urakkavuosilta.
Nyt dialogi fiksattiin kutienkin siten, että siihen tulee hintoja vain
nykyiseltä urakkavuodelta. Ei ole järkeä näyttää vuosilukuja, kun ne ovat
kuitenkin kaikissa vaihtoehdoissa samat.
